src/pages/home/report/ReportActionCompose/SuggestionMention.tsx

```diff
@@ -270,9 +271,19 @@ function SuggestionMention(
                 }
 
                 return true;
-            });
+            }) as Array<PersonalDetails & {weight: number}>; // at this point we are sure that the details are not null
+
+            const sortedPersonalDetails = filteredPersonalDetails
+                .map((user) => ({
+                    originalDetail: user,
+                    weight: user.weight,
+                    accountID: user.accountID,
+                    // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+                    displayName: ReportUtils.getDisplayNameForParticipant(user.accountID) || user.login || '',
+                }))
+                .sort((first, second) => compareUserInList(first, second))
+                .map((userAfterCompare) => userAfterCompare.originalDetail);
```

Comment:
I don't like this mapping. How about moving first map to sort function? Then 2nd map is redundant.